From fe82e9d73a02bec187a01bdf7e377f14a96adca6 Mon Sep 17 00:00:00 2001
From: Lasse Brun <bruners@gmail.com>
Date: Sat, 4 May 2013 00:45:43 +0200
Subject: [PATCH] Patch to fix linking with libpthread
Source: Based on patches from gentoo (#367607, *kash-link-pthread.patch)
Upstream: No info, 0.1.5-p2 is 3 years behind current upstream which builds fine without

---
 src/kash/Makefile.kmk | 1 +
 src/kmk/Makefile.kmk  | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/kash/Makefile.kmk b/src/kash/Makefile.kmk
index aca65f6..2016988 100644
--- a/src/kash/Makefile.kmk
+++ b/src/kash/Makefile.kmk
@@ -52,6 +52,7 @@ kash_INCS = $(PATH_kash) . # (the last is because of error.h)
 if "$(USER)" == "bird" && "$(KBUILD_TARGET)" != "win"
 kash_CFLAGS += -std=gnu99
 endif
+kash_LIBS += pthread
 kash_SOURCES = \
 	main.c \
 	alias.c \
diff --git a/src/kmk/Makefile.kmk b/src/kmk/Makefile.kmk
index 12b9697..2829ccc 100644
--- a/src/kmk/Makefile.kmk
+++ b/src/kmk/Makefile.kmk
@@ -58,7 +58,7 @@ TEMPLATE_BIN-KMK_INCS.darwin = glob
 TEMPLATE_BIN-KMK_INCS.dragonfly = glob
 TEMPLATE_BIN-KMK_INCS.freebsd = glob
 TEMPLATE_BIN-KMK_INCS.solaris = glob
-TEMPLATE_BIN-KMK_LIBS = $(TEMPLATE_BIN_LIBS) $(TARGET_kmkmissing) $(LIB_KUTIL)
+TEMPLATE_BIN-KMK_LIBS = $(TEMPLATE_BIN_LIBS) $(TARGET_kmkmissing) $(LIB_KUTIL) pthread
 ifdef ELECTRIC_HEAP # for electric heap (see electric.c) - windows only.
 ifeq ($(KBUILD_TARGET),win)
  TEMPLATE_BIN-KMK_CFLAGS = $(TEMPLATE_BIN_CFLAGS) /FI$(kmk_PATH)/electric.h -DELECTRIC_HEAP=1
-- 
1.8.2.1
